src/mlte/measurement/process_measurement.py

```diff
@@ -0,0 +1,94 @@
+"""
+Base class for measurement of external processes asynchronally.
+"""
+
+from __future__ import annotations
+
+import threading
+import time
+from typing import List
+
+from .measurement import Measurement
+from .result import Result
+from mlte._private.platform import is_windows
+from mlte._private import job
+
+# -----------------------------------------------------------------------------
+# ProcessMeasurement
+# -----------------------------------------------------------------------------
+
+class ProcessMeasurement(Measurement):
+    """Base class to be extended to measure external processes."""
+
+    @staticmethod
+    def start_process(script: str, arguments: List[str]) -> int:
+        """
+        Initialize an external process running training or similar script.
+
+        :param script: The full path to a Python script with the training or equivalent process to run.
+        :type script: str
+
+        :param arguments: A list of string arguments for the process.
+        :type arguments: List[str[]
+
+        :return: the id of the process that was created.
+        :rtype: int
+        """        
+        return job.spawn_python_job(script, arguments)
+
+    def __init__(self, instance: ProcessMeasurement, identifier: str):
+        """
+        Initialize a new ProcessMeasurement measurement.
+
+        :param identifier: A unique identifier for the measurement
+        :type identifier: str
+        """
+        super().__init__(instance, identifier)        
+        if is_windows():
+            raise RuntimeError(
+                f"Measurement {self.name} is not supported on Windows."
+            )
+        self.thread = None
+        self.result = None
+        self.error = None
+
+    def evaluate_async(self, pid: int, *args, **kwargs):
+        """
+        Monitor an external process at `pid` in a separate thread until it stops.
+        Equivalent to evaluate(), but does not return the result immediately as it works in the background.
+
+        :param pid: The process identifier
+        :type pid: int
+        """
+
+        # Evaluate the measurement
+        self.error = None
+        self.result = None
+        self.thread = threading.Thread(target=lambda: self._run_call(pid, *args, **kwargs))
+        self.thread.start()
+ 
+    def _run_call(self, pid, *args, **kwargs):
+        """
+        Runs the internall __call__ method that should implement the measurement, and stores its results when it finishes.
+        """
+        try:
+            self.result = self.__call__(pid, *args, **kwargs)
+        except Exception as e:
+            self.error = "Could not evaluate process: " + str(e)
+
+    def wait_for_result(self) -> Result:
```

Comment:
We should consider adding some mechanism to wait for all asynchronous results in a collection. This can be a separate feature / pull-request through.

Reply:
I agree that we should do that (it could even be with Python's futures and executor), but I also agree this can be a separate feature since it could need more work or change a bit the structure.